src/lib/accessibility/FocusZone/FocusZone.tsx

```diff
@@ -0,0 +1,992 @@
+import {
+  FocusZoneDirection,
+  FocusZoneTabbableElements,
+  IFocusZone,
+  IFocusZoneProps,
+} from './FocusZone.types'
+import {
+  BaseComponent,
+  EventGroup,
+  KeyCodes,
+  css,
+  htmlElementProperties,
+  elementContains,
+  getDocument,
+  getId,
+  getNextElement,
+  getNativeProps,
+  getParent,
+  getPreviousElement,
+  getRTL,
+  isElementFocusZone,
+  isElementFocusSubZone,
+  isElementTabbable,
+  shouldWrapFocus,
+  createRef,
+} from '@uifabric/utilities'
+import { IS_FOCUSABLE_ATTRIBUTE } from '../interfaces'
+
+const IS_ENTER_DISABLED_ATTRIBUTE = 'data-disable-click-on-enter'
+export const FOCUSZONE_ID_ATTRIBUTE = 'data-focuszone-id'
+const TABINDEX = 'tabindex'
+const NO_VERTICAL_WRAP = 'data-no-vertical-wrap'
+const NO_HORIZONTAL_WRAP = 'data-no-horizontal-wrap'
+const LARGE_DISTANCE_FROM_CENTER = 999999999
+const LARGE_NEGATIVE_DISTANCE_FROM_CENTER = -999999999
+
+const _allInstances: {
+  [key: string]: FocusZone
+} = {}
+
+interface IPoint {
+  left: number
+  top: number
+}
+const ALLOWED_INPUT_TYPES = ['text', 'number', 'password', 'email', 'tel', 'url', 'search']
+
+const ALLOW_VIRTUAL_ELEMENTS = false
+
+export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFocusZone {
+  static propTypes = {
+    isCircularNavigation: PropTypes.bool,
+    direction: PropTypes.number,
+  }
+
+  public static defaultProps: IFocusZoneProps = {
+    isCircularNavigation: false,
+    direction: FocusZoneDirection.bidirectional,
+  }
+
+  private _root = createRef<HTMLElement>()
+  private _id: string
+  /** The most recently focused child element. */
+  private _activeElement: HTMLElement | null
+  /** The child element with tabindex=0. */
+  private _defaultFocusElement: HTMLElement | null
+  private _focusAlignment: IPoint
+  private _isInnerZone: boolean
+
+  /** Used to allow us to move to next focusable element even when we're focusing on a input element when pressing tab */
+  private _processingTabKey: boolean
+
+  constructor(props: IFocusZoneProps) {
+    super(props)
+
+    this._warnDeprecations({
+      rootProps: undefined,
+      allowTabKey: 'handleTabKey',
+    })
+
+    this._id = getId('FocusZone')
+
+    this._focusAlignment = {
+      left: 0,
+      top: 0,
+    }
+
+    this._processingTabKey = false
+  }
+
+  public componentDidMount(): void {
+    _allInstances[this._id] = this
+    if (this._root.current) {
+      const windowElement = this._root.current.ownerDocument.defaultView
+
+      let parentElement = getParent(this._root.current, ALLOW_VIRTUAL_ELEMENTS)
+
+      while (parentElement && parentElement !== document.body && parentElement.nodeType === 1) {
+        if (isElementFocusZone(parentElement)) {
+          this._isInnerZone = true
+          break
+        }
+        parentElement = getParent(parentElement, ALLOW_VIRTUAL_ELEMENTS)
+      }
+
+      if (!this._isInnerZone) {
+        this._events.on(windowElement, 'keydown', this.onKeyDownCapture, true)
```

Comment:
It is not clear where `this._events` is defined, however, we also have an `eventStack` util which is likely doing something very similar to `this._events` here.  Concerned with the duplication this would introduce.